### PR TITLE
Add pipeline.embed

### PR DESCRIPTION
**Pull request by abdulfatir**
*Description of changes:* This PR adds `pipeline.embed` which extracts encoder embeddings from the model. These embeddings may be useful for some downstream tasks such as classification, so this is useful to have. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/chronos/chronos.py [hunk 1]
```diff
@@ -209,6 +209,37 @@ def __init__(self, config: ChronosConfig, model: PreTrainedModel) -> None:
     def device(self):
         return self.model.device
 
+    @torch.no_grad()
```
Comment (lostella) [inline src/chronos/chronos.py:212]: I'm wondering: should this be enforced here, or rather at invocation time?

File: src/chronos/chronos.py [hunk 2]
```diff
@@ -306,6 +337,49 @@ def __init__(self, tokenizer, model):
...
+        if context.ndim == 1:
+            context = context.unsqueeze(0)
+        assert context.ndim == 2
+
+        return context
+
+    def embed(
+        self, context: Union[torch.Tensor, List[torch.Tensor]]
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        """Get encoder embeddings for the given time series.
```
Comment (lostella) [inline src/chronos/chronos.py:355]:
```suggestion
        """
        Get encoder embeddings for the given time series.
```

File: src/chronos/chronos.py (hunk 1 above)
Comment (abdulfatir) [reply src/chronos/chronos.py]: Moved to `pipeline.embed` instead.

File: src/chronos/chronos.py [hunk 3]
```diff
@@ -209,6 +209,36 @@ def __init__(self, config: ChronosConfig, model: PreTrainedModel) -> None:
     def device(self):
         return self.model.device
 
+    def encode(
+        self,
+        input_ids: torch.Tensor,
+        attention_mask: torch.Tensor,
+    ):
```
Comment (lostella) [inline src/chronos/chronos.py:216]: Just thinking out loud: we could use this also to optimize "batch sampling" (e.g. sampling 20 at a time, five times to get 100 samples)

File: src/chronos/chronos.py [hunk 4]
```diff
@@ -306,6 +336,51 @@ def __init__(self, tokenizer, model):
...
+            ``torch.nan`` to align series of different lengths.
+
+        Returns
+        -------
+        embeddings, scale
+            A tuple of two tensors: the encoder embeddings and the time series scale.
+            The encoder embeddings are shaped (batch_size, context_length, d_model)
+            or (batch_size, context_length + 1, d_model), where the extra 1 is for EOS.
+            If your original time series were shorter than the model's context_length,
+            please slice the returned embeddings along the time axis accordingly.
```
Comment (lostella) [inline src/chronos/chronos.py:373]: Maybe this sentence can be remove if #25 is merged

**Review by lostella — COMMENTED** [review]
Should we bump the package version number to 1.1?

File: src/chronos/chronos.py [hunk 5]
```diff
@@ -306,6 +336,51 @@ def __init__(self, tokenizer, model):
...
+            The scale is shaped (batch_size, ).
+        """
+        context = self._prepare_and_validate_context(context=context)
+        token_ids, attention_mask, scale = self.tokenizer.input_transform(context)
+        embeddings = self.model.encode(
+            input_ids=token_ids.to(self.model.device),
+            attention_mask=attention_mask.to(self.model.device),
+        ).cpu()
```
[the remark this answers is not in the document]
Comment (abdulfatir) [reply src/chronos/chronos.py:381]: Because the predictions are also sent to CPU.

File: src/chronos/chronos.py (hunk 3 above)
Comment (abdulfatir) [reply src/chronos/chronos.py:216]: I'm not sure I get it. You mean "caching" the embeddings?

File: src/chronos/chronos.py (hunk 4 above)
Comment (abdulfatir) [reply src/chronos/chronos.py:373]: Yes.

**Comment by abdulfatir** [comment]
> Should we bump the package version number to 1.1?

Yes, we should.

File: src/chronos/chronos.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (abdulfatir) [reply src/chronos/chronos.py:381]: Should I keep it on device?

File: src/chronos/chronos.py [hunk 6]
```diff
@@ -306,6 +336,51 @@ def __init__(self, tokenizer, model):
...
+
+        Returns
+        -------
+        embeddings, scale
+            A tuple of two tensors: the encoder embeddings and the time series scale.
+            The encoder embeddings are shaped (batch_size, context_length, d_model)
+            or (batch_size, context_length + 1, d_model), where the extra 1 is for EOS.
+            If your original time series were shorter than the model's context_length,
+            please slice the returned embeddings along the time axis accordingly.
+            The scale is shaped (batch_size, ).
```
[the remark this answers is not in the document]
Comment (abdulfatir) [reply src/chronos/chronos.py]: - Renamed.
- They are not. Currently they are 512 + 1 on the time axis.

**Comment by abdulfatir** [comment]
Rolled back version bump. I think we should do that after both open PRs have been merged.

File: src/chronos/chronos.py (hunk 4 above)
Comment (abdulfatir) [reply src/chronos/chronos.py]: Done.

**Review by lostella — APPROVED** [review]
